File: 3_Semestre/Estrutura_Dados/Aula01/exercicio03.java
package Aula01;

import java.util.Arrays;
import java.util.Random;

// Considere uma matriz bidimensional de ordem igual a cinco com valores reais de simples precisão.
// Leia a matriz, via teclado, e na saída imprima todos os valores da diagonal principal.

public class exercicio03 {
    public static void main(String[] args) {
        int[][] vetor = new int[5][5];

        gerarNum(vetor, 0, 100);
        imprimirVetor(vetor);
    }

    public static void gerarNum(int[][] vetor, int min, int max) {
        Random rd = new Random();

        for(int i = 0; i < vetor.length; i++) {
            for(int j = 0; j < vetor[i].length; j++) {
                vetor[i][j] = rd.nextInt(max - min) + min;
            }
        }
    }

    public static void imprimirVetor(int[][] vetor) {
        int temp = 0;
        
        System.out.println("Array: ");
        for(int i = 0; i < vetor.length; i++) {
            System.out.println(Arrays.toString(vetor[i]));
            for(int j = 0; j < vetor[i].length; j++) {
                if (i == j) {
                    temp += vetor[i][j];
                }
            }
        }

        System.out.println("Soma diagonal principal: " + temp);
    }
}

File: 3_Semestre/Estrutura_Dados/Aula01/exercicio03.js
function criarArray(coluna, linha) {
    return Array.from(Array(coluna), () => new Array(linha))
}

function randomNum(min, max) {
    min = Math.ceil(min)
    max = Math.floor(max)

    return Math.floor(Math.random() * (max - min) + min)
}

function gerarNum(array, min, max) {
    for(var i = 0; i < array.length; i++) {
        for(var j = 0; j < array[i].length; j++) {
            array[i][j] = randomNum(min, max)
        }
    }
}

function somaDiagonal(array) {
    var temp = 0
    
    for (var i = 0; i < array.length; i++) {
        for (var j = 0; j < array[i].length; j++) {
            
            if (i == j) {
                temp += array[i][j]
            }
        }
    }

    console.log('Soma diagonal: ' + temp)
}

var teste = criarArray(5, 5)
gerarNum(teste, 0, 100)
somaDiagonal(teste)
console.log(teste)